src/client/GameWindow.cpp

```diff
@@ -400,47 +439,73 @@ bool GameWindow::priorityCardClickAction(sf::Vector2i clickPosition)
  * @param clickPosition is the position on the cursor when the user click
  * @brief Dectect click and actions to do after
  */
-void GameWindow::clickAction(sf::Vector2i clickPosition)
+bool GameWindow::clickAction(sf::Event& event, sf::Vector2i clickPosition, std::shared_ptr<bool> moveMode)
 {
     int minimumDistance = WINDOW_LENGTH;
     std::array<int, 2> hexagonOnClick = {0, 0};
 
     bool isClickable = false;
 
-    if (priorityCardClickAction(clickPosition)) {
-            return;
-    }
-
-    for (auto &mapTexture : mapTextureToDisplay)
+    if (!*moveMode)
     {
-        for (unsigned j = 0; j < mapTexture.getSize(); j++)
-        { 
-            if (gameEnginePtr->intersectPointRect(clickPosition, mapTexture.getSprite(j).getGlobalBounds()))
+        if (priorityCardClickAction(clickPosition)) {
+            return false;
+        }
+
+        for (auto &mapTexture : mapTextureToDisplay)
```

Comment:
At some point we will really need a function for that, the `if (!*moveMode)` add another level of indentation